### PR TITLE
[ALLUXIO-2112] Refactor property setting in HadoopUtils.java

### PR DESCRIPTION
**Pull request by MichaelPhantom**
[ALLUXIO-2112](https://alluxio.atlassian.net/browse/ALLUXIO-2112)
Guys, refactored according to my knowledge, please help review it, any issue notice me, thanks.

### CODE REVIEW

**Comment by gpang** [comment]
ok to test

**Comment by gpang** [comment]
Jenkins, add to whitelist

File: core/client/src/main/java/alluxio/hadoop/HadoopUtils.java [hunk 1]
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
...
+    for (final String propertyName : propertyNames) {
+      setProperty(configuration, propertyName);
     }
-    String swiftSimulationConf = Constants.SWIFT_SIMULATION;
-    if (System.getProperty(swiftSimulationConf) != null
-        && configuration.get(swiftSimulationConf) == null) {
-      configuration.set(swiftSimulationConf, System.getProperty(swiftSimulationConf));
+  }
+
+  private static void setProperty(final Configuration configuration, final String propertyName) {
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/HadoopUtils.java]: add javadoc that this method won't override existing properties even if they are set as System properties.

File: core/client/src/main/java/alluxio/hadoop/HadoopUtils.java [hunk 2]
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
...
-    if (System.getProperty(passwordKeyConf) != null
-        && configuration.get(passwordKeyConf) == null) {
-      configuration.set(passwordKeyConf, System.getProperty(passwordKeyConf));
+    final String[] propertyNames = {Constants.SWIFT_API_KEY, Constants.SWIFT_TENANT_KEY,
+      Constants.SWIFT_USER_KEY, Constants.SWIFT_AUTH_URL_KEY, Constants.SWIFT_AUTH_METHOD_KEY,
+      Constants.SWIFT_PASSWORD_KEY, Constants.SWIFT_SIMULATION};
+    setProperty(configuration, propertyNames);
+  }
+
+  private static void setProperty(final Configuration configuration, final String[] propertyNames) {
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/HadoopUtils.java]: maybe call this `setConfigurationFromSystemProperties` and call the other method `setConfigurationFromSystemProperty`?

File: core/client/src/main/java/alluxio/hadoop/HadoopUtils.java [hunk 3]
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
...
-    String authMethodKeyConf = Constants.SWIFT_AUTH_METHOD_KEY;
-    if (System.getProperty(authMethodKeyConf) != null
-        && configuration.get(authMethodKeyConf) == null) {
-      configuration.set(authMethodKeyConf, System.getProperty(authMethodKeyConf));
-    }
-    String passwordKeyConf = Constants.SWIFT_PASSWORD_KEY;
-    if (System.getProperty(passwordKeyConf) != null
-        && configuration.get(passwordKeyConf) == null) {
-      configuration.set(passwordKeyConf, System.getProperty(passwordKeyConf));
+    final String[] propertyNames = {Constants.SWIFT_API_KEY, Constants.SWIFT_TENANT_KEY,
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/HadoopUtils.java]: we typically don't mark local variables as `final`

**Comment by aaudiber** [comment]
Thanks for the refactoring @MichaelPhantom! I've left a few comments in the code

**Comment by MichaelPhantom** [comment]
@aaudiber thank you so much for the detail comments, last commit removed final decorator on local variables, renamed method names and added java document accordingly.

File: core/client/src/main/java/alluxio/hadoop/HadoopUtils.java [hunk 4]
```diff
@@ -164,46 +156,44 @@ public static void addS3Credentials(Configuration conf) {
...
+
+  /**
+   * Set the System properties into Hadoop configuration.
+   *
+   * This method won't override existing properties even if they are set as System properties.
+   *
+   * @param configuration Hadoop configuration
+   * @param propertyNames the properties to be set
+   */
+  private static void setConfigurationFromSystemProperties(final Configuration configuration,
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/HadoopUtils.java]: we can remove the final keyword for the method parameters as well, here and below

**Comment by aaudiber** [comment]
@MichaelPhantom I left one last minor comment, otherwise looks good to me!

**Comment by MichaelPhantom** [comment]
@aaudiber the latest commit removed the final decorators as commented, thanks.

**Comment by aaudiber** [comment]
LGTM, thanks for the iterations

**Pull request merged**